src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ArrayOfFrozenObjectsNode.cs

```diff
@@ -38,8 +61,20 @@ protected override void GetElementDataForNodes(ref ObjectDataBuilder builder, No
             // Terminate with a null pointer as expected by the GC
             AlignNextObject(ref builder, factory);
             builder.EmitZeroPointer();
+
+            _endSymbol.SetSymbolOffset(builder.CountBytes);
+            builder.AddSymbol(_endSymbol);
+
+            return builder.ToObjectData();
         }
 
+        protected override ObjectNodeSection GetDehydratedSection(NodeFactory factory) => ObjectNodeSection.DataSection;
+        protected override string GetName(NodeFactory factory) => this.GetMangledName(factory.NameMangler);
+
         public override int ClassCode => -1771336339;
+
+        public override bool IsShareable => false;
```

Comment:
We have a (mostly test-only) mode where each managed assembly can be compiled into a separate object file, and then linked into an EXE by the linker.

In this mode, some nodes can end up being duplicated (e.g. `List<int>.Add`) because we don't know if any other object file could be defining that (the compilation is done in isolation). We mark such nodes as shareable. Object writing will place them into a [COMDAT section](https://stackoverflow.com/questions/1834597/what-is-the-comdat-section-used-for) and that allows linker to pick any definition (it would error out otherwise).